client/src/main/java/io/pravega/client/stream/EventPointer.java

```diff
@@ -45,5 +45,12 @@
     static EventPointer fromBytes(ByteBuffer eventPointer) {
         return EventPointerInternal.fromBytes(eventPointer);
     }
-    
+
+    /**
+     * Returns the stream name from which event is read from , If no event to read then this will return null.
+     *
+     * @return The name of the stream
+     */
+    String getStreamName();
```

Comment:
I think it might be better to return the Stream object. That is a public class already.